02-Context-or-Narrative/State-of-PureScript-Documentation-2019.md

```diff
@@ -1,28 +1,37 @@
 # State of PureScript Documentation: 2019
 
+<!-- https://magnetikonline.github.io/markdown-toc-generate/ -->
+
+- [Why Read This Document?](#why-read-this-document)
+- [What is "Good" Documentation Anyway?](#what-is-good-documentation-anyway)
+    - [The Types of Documentation](#the-types-of-documentation)
+    - [The Documentation's Intended Audience](#the-documentations-intended-audience)
+    - [Maintaining Documentation's Accuracy](#maintaining-documentations-accuracy)
+        - [The "Size" of a Change](#the-size-of-a-change)
+        - [The "Frequency" of a Change](#the-frequency-of-a-change)
+        - [How to Make Maintenance Easier](#how-to-make-maintenance-easier)
+    - [Criteria for "Good" Documentation](#criteria-for-good-documentation)
+    - [Evaluating PureScript's Documentation](#evaluating-purescripts-documentation)
+- [Why is PureScript's Documentation Lacking and How Do We Improve It?](#why-is-purescripts-documentation-lacking-and-how-do-we-improve-it)
+- [New Learners: What is the Best Way to Learn PureScript?](#new-learners-what-is-the-best-way-to-learn-purescript)
+- [PureScript Documentation Writers: What is the Best Way to Write Documentation in this Context?](#purescript-documentation-writers-what-is-the-best-way-to-write-documentation-in-this-context)
+- [Core Contributors: ???](#core-contributors-)
+
```

Comment:
I like that this will have a ToC section. However, I wonder whether this should go after the 'why read this document' section. Otherwise, that section isn't the first thing people see and they might think it's just going to be a massive ToC before the get to the real content, so they leave the page and go elsewhere.

Reply:
I like that idea. I removed the "## Why Read This Document?" section because, in my experience, the TOC is usually in the very top section to introduce a navigation to and overview of all following sections. LMK what you think about this change.

Reply:
As long as we keep the 'why read this document' section brief, I think both concerns are still adequately addressed (the navigational "norm" is still a principle to which we adhere, and people have small enough thing to read that they do not immediately navigate away when they shouldn't)